core.ts

```diff
@@ -977,7 +978,87 @@ abstract class PinejsClientCoreTemplate<
 		return this.request(params, { method: 'DELETE' });
 	}
 
-	compile(params: PinejsClientCoreFactory.Params) {
+	prepare<
+		T extends {
+			[index: string]: null | string | number | boolean | Date;
+		}
+	>(
+		params: string | (PinejsClientCoreFactory.ParamsObj & { method?: 'GET' }),
+	): PreparedFn<T, PromiseResult>;
+	prepare<
+		T extends {
+			[index: string]: null | string | number | boolean | Date;
+		}
+	>(
+		params: PinejsClientCoreFactory.ParamsObj & {
+			method: Exclude<PinejsClientCoreFactory.ParamsObj['method'], 'GET'>;
+		},
+	): PreparedFn<T, PromiseObj>;
+	prepare<
+		T extends {
+			[index: string]: null | string | number | boolean | Date;
+		}
+	>(
+		params: PinejsClientCoreFactory.Params,
+	): PreparedFn<T, PromiseObj | PromiseResult> {
```

Comment:
This signature is insane :) Can it be broken out to type aliases to make it more readable?

Reply:
I've done what I can but there's a limit based on the fact that by necessity the types need to be different or otherwise they'd no longer be overloads :P